packages/jest-cli/src/__tests__/cli/args.test.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+'use strict';
+
+import type {Argv} from 'types/Argv';
+import {check} from '../../cli/args';
+
+describe('check', () => {
+  it('returns true if the arguments are valid', () => {
+    const argv: Argv = {};
+    expect(check(argv)).toBe(true);
+  });
+
+  it('raises an exception if runInBand and maxWorkers are both specified', () => {
+    const argv: Argv = {maxWorkers: 2, runInBand: true};
+    expect(() => check(argv)).toThrow();
+  });
+
+  it('raises an exception if onlyChanged and watchAll are both specified', () => {
+    const argv: Argv = {onlyChanged: true, watchAll: true};
+    expect(() => check(argv)).toThrow();
+  });
+
+  it('raises an exception if findRelatedTests is specified with no file paths', () => {
+    const argv: Argv = {findRelatedTests: true};
+    expect(() => check(argv)).toThrow();
+  });
+
+  it('raises an exception if maxWorkers is specified with no number', () => {
+    const argv: Argv = {maxWorkers: undefined};
+    expect(() => check(argv)).toThrow();
+  });
+
+  it('raises an exception if config is not a valid JSON string', () => {
+    const argv: Argv = {config: 'x:1'};
+    expect(() => check(argv)).toThrow();
```

Comment:
Tests, nice! I think we should at least check the core message of the error though.

Something like this would be more descriptive:
```js
expect(() => check(argv)).toThrowError('not a valid JSON string');
```